hazelcast/src/test/java/com/hazelcast/test/bounce/BounceMemberRule.java

```diff
@@ -363,25 +388,33 @@ public void evaluate() throws Throwable {
 
     // wait until all test tasks complete or one of them throws an exception
     private void waitForFutures(Future[] futures) {
+        // do not wait more than 30 seconds
+        long deadline = currentTimeMillis() + 30 * 1000;
```

Comment:
I would make the timout a constant like `TIMEOUT_SECONDS` and use `TimeUnit.Seconds.toMillis(TIMEOUT_SECONDS)` here (and of course use the constant in the comment and warning log below as well).